test/Semantics/select-rank.f90

```diff
@@ -0,0 +1,266 @@
+! RUN: %S/test_errors.sh %s %flang %t
+!Tests for SELECT RANK Construct(R1148)
+program select_rank
+   implicit none
+   integer, dimension(10:30, 10:20, -1:20) :: x
+   integer, parameter :: y(*) = [1,2,3,4]
+   integer, dimension(5) :: z
+   integer, allocatable :: a(:)
+
+   allocate(a(10:20))
+
+   call CALL_SHAPE(x)
+   call CALL_SHAPE(y)
+   call CALL_SHAPE(z)
+   call CALL_SHAPE(a)
+
+contains
+   !No error expected
+   subroutine CALL_ME(x)
+    implicit none
+    integer :: x(..)
+    SELECT RANK(x)
+    RANK (0)
+      print *, "PRINT RANK 0"
+    RANK (1)
+      print *, "PRINT RANK 1"
+    END SELECT
+   end
+
+   subroutine CALL_ME9(x)
+    implicit none
+    integer :: x(..),j
+    boo: SELECT RANK(x)
+    RANK (1+0)
+      print *, "PRINT RANK 1"
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == (1+0)))
+    END SELECT boo
+   end subroutine
+
+   !Error expected
+   subroutine CALL_ME2(x)
+    implicit none
+    integer :: x(..)
+    integer :: y(3),j
+    !ERROR: Selector 'y' is not an assumed-rank array variable
+    SELECT RANK(y)
+    RANK (0)
+      print *, "PRINT RANK 0"
+    RANK (1)
+      print *, "PRINT RANK 1"
+     END SELECT
+
+    SELECT RANK(x)
+    RANK(0)
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 0)) ! will fail when RANK(x) is not zero here
+    END SELECT
+   end subroutine
+
+   subroutine CALL_ME3(x)
+    implicit none
+    integer :: x(..),j
+    SELECT RANK(x)
+    !ERROR: The value of the selector must be between zero and 15
+    RANK (16)
+      j = INT(0, KIND=MERGE(KIND(0), -1, RANK(x) == 16))
+    END SELECT
```

Comment:
What about errors in type (`RANK(.TRUE.)` or `RANK(1.0)`) or rank (`RANK(array)`) or variability (`RANK(x)` with non-constant x)?

Reply:
They are already caught by `C713`.
Thanks @klausler I have added tests.